streamz/tests/test_core.py

```diff
@@ -1063,5 +1064,23 @@ def test_share_common_ioloop(clean):
     assert aa.loop is bb.loop
 
 
+@pytest.mark.parametrize('delimiter', ['\n', '=='])
+def test_from_text(delimiter):
+    with tmpfile() as fn:
+        open(fn, 'w')  # touch file - needs to exist
+        stream = Stream.from_textfile(fn, delimiter=delimiter,
+                                      poll_interval=0.01)
+        l = stream.sink_to_list()
+        open(fn, 'a').write('notaline')
+        sleep(0.05)
+        assert l == []
+        open(fn, 'a').write(delimiter)
+        sleep(0.05)
+        assert l == ['notaline' + delimiter]
```

Comment:
This may not trigger on slow CI systems, leading to intermittent errors.  I recommend polling while this is not true and then asserting that some wildly pessimistic timeout has not passed.  Something like the following:

```python
start = time()
while L != ['notaline' + delimiter]:
    sleep(0.01)
    assert time() < start + 2
```

Reply:
This pattern is too common, it should be in test_utils; but in this case, this is only really like a `yield`, i.e., coroutines should run just once.

Reply:
I think that Antoine put something like this in `distributed/utils.py::wait_for`.  In practice I find that I more often just do the while-sleep mechanism above, but I'm happy with either.